honeybadgermpc/passive.py

```diff
@@ -136,6 +138,32 @@ def write_shares(f, modulus, degree, myid, shares):
 
 
 def shareInContext(context):
+
+    def _binopField(fut, other, op):
+        assert type(other) in [ShareFuture, FieldFuture, Share, Field]
+        if isinstance(other, ShareFuture) or isinstance(other, Share):
+            res = ShareFuture()
+        elif isinstance(other, Field) or isinstance(other, FieldFuture):
+            res = FieldFuture()
+
+        if isinstance(other, Future):
+            def cb(_): return res.set_result(op(fut.result(), other.result()))
+            asyncio.gather(fut, other).add_done_callback(cb)
+        else:
+            def cb(_): return res.set_result(op(fut.result(), other))
+            fut.add_done_callback(cb)
+        return res
+
+    class FieldFuture(Future):
```

Comment:
How about a class derived from `Future` like `BaseFuture` and then `FieldFuture` and `ShareFuture` derive from `BaseFuture`?

Will this help us avoid having the same code (the overloading of the operators) in two separate places?